app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +135,196 @@ object DialogUtils {
+            gradientPicker.setHue(hsvState.hue)
+            gradientPicker.setSaturationAndValue(hsvState.saturation, hsvState.value)
+            topColorSquare.setBackgroundColor(parseColor(hex))
+        }
+        val colorState = ColorUtils.calculateColorWithHsv(previousColor)
+        forceViewUpdates(colorState)
+        bottomColorSquare.setBackgroundColor(parseColor(colorState.hex))
+        /**
+         * Function that clears focus on whatever view the user has currently selected.
+         * Currently only called when the user interacts with the Gradient or Hue Slider.
+         */
+        fun clearFocus() {
+            listOf(hexText, hueText, satText, valText, redText, greenText, blueText).forEach {
+                it.clearFocus()
+            }
+        }
+        listOf(redText, greenText, blueText).forEach {
+            it.addTextChangedListener {
+                val calculatedColorState = ColorUtils.calculateColorWithRgb(
+                    intArrayOf(
+                        redText.rgbIntConvert,
+                        greenText.rgbIntConvert,
+                        blueText.rgbIntConvert
+                    )
+                )
+                forceViewUpdates(calculatedColorState)
+                onUpdate.invoke(
+                    floatArrayOf(
+                        hueText.floatConvert, satText.floatConvert, valText.floatConvert
+                    )
+                )
+                huePicker.setHue(calculatedColorState.hue)
+                gradientPicker.setSaturationAndValue(
+                    calculatedColorState.saturation,
+                    calculatedColorState.value
+                )
+                forceViewUpdates(calculatedColorState)
+                onUpdate.invoke(
+                    floatArrayOf(
+                        calculatedColorState.hue,
+                        calculatedColorState.saturation,
+                        calculatedColorState.value
+                    )
+                )
+            }
+        }
+        listOf(hueText, satText, valText).forEach {
+            it.addTextChangedListener {
+                val hsv = floatArrayOf(
+                    hueText.floatConvert,
+                    satText.floatConvert / 100,
+                    valText.floatConvert / 100
+                )
+                val calculatedColorState = ColorUtils.calculateColorWithHsv(hsv)
+                huePicker.setHue(calculatedColorState.hue)
+                gradientPicker.setSaturationAndValue(
+                    calculatedColorState.saturation,
+                    calculatedColorState.value
+                )
+                forceViewUpdates(calculatedColorState)
+                onUpdate.invoke(hsv)
+            }
+        }
+        huePicker?.apply {
+            onHueChanged = { hue ->
+                if (satText.colorInt != null && valText.colorInt != null
+                ) {
+                    val hsv = floatArrayOf(
```

Comment:
Use your Hsv object for these instead of float arrays

Reply:
using hsv object